au4/src/au3wrap/iaudacity3audiooutput.h

```diff
@@ -0,0 +1,31 @@
+/*
+* Audacity: A Digital Audio Editor
+*/
+
+#ifndef AU_AU3WRAP_IAUDACITYAUDIOOUTPUT_H
+#define AU_AU3WRAP_IAUDACITYAUDIOOUTPUT_H
+
+#include <memory>
+
+#include "global/async/promise.h"
+#include "global/async/channel.h"
+
+#include "playback/audiotypes.h"
+
+namespace au::au3 {
+class IAudacity3AudioOutput
+{
+public:
+    virtual ~IAudacity3AudioOutput() = default;
+
+    virtual muse::async::Promise<float> playbackVolume() const = 0;
```

Comment:
Hmm, is Promise really appropriate here?
Typically, a promise is used in cases where you call some kind of asynchronous action, and you need to get the result of this action - successful, not successful

Reply:
ahh, I get it, this is an asynchronous retrieval of the value... ok.

Reply:
Actually, the idea is to make the interfaces of au3 classes and similar classes from the framework the same, so that it is easier to replace later